backpack/core/derivatives/avgpoolnd.py

```diff
@@ -0,0 +1,179 @@
+"""The code relies on the insight that average pooling can be understood as
+convolution over single channels with a constant kernel."""
+
+import torch.nn
+from torch.nn import (
+    Conv1d,
+    Conv2d,
+    Conv3d,
+    ConvTranspose1d,
+    ConvTranspose2d,
+    ConvTranspose3d,
+)
+
+from backpack.core.derivatives.basederivatives import BaseDerivatives
+from backpack.utils.ein import eingroup
+
+
+class AvgPoolNDDerivatives(BaseDerivatives):
+    def __init__(self, N):
+        self.N = N
+        if self.N == 1:
+            self.conv = Conv1d
+            self.convt = ConvTranspose1d
+            self.equation = "v,n,c,l->vnc,l"
+        elif self.N == 2:
+            self.conv = Conv2d
+            self.convt = ConvTranspose2d
+            self.equation = "v,n,c,w,h->vnc,w,h"
+        elif self.N == 3:
+            self.conv = Conv3d
+            self.convt = ConvTranspose3d
+            self.equation = "v,n,c,d,w,h->vnc,d,w,h"
+        else:
+            raise ValueError("{}-dimensional Conv. is not implemented.".format(self.N))
+
+    def hessian_is_zero(self):
+        return True
+
+    def ea_jac_t_mat_jac_prod(self, module, g_inp, g_out, mat):
+        """Use fact that average pooling can be implemented as conv."""
+        if self.N == 1:
+            _, C, L_in = module.input0.size()
+            _, _, L_out = module.output.size()
+            in_features = C * L_in
+            out_features = C * L_out
+            shape_out = (1, L_out)
+        elif self.N == 2:
+            _, C, H_in, W_in = module.input0.size()
+            _, _, H_out, W_out = module.output.size()
+            in_features = C * H_in * W_in
+            out_features = C * H_out * W_out
+            shape_out = (1, H_out, W_out)
+        elif self.N == 3:
+            _, C, D_in, H_in, W_in = module.input0.size()
+            _, _, D_out, H_out, W_out = module.output.size()
+            in_features = C * D_in * H_in * W_in
+            out_features = C * D_out * H_out * W_out
+            shape_out = (1, D_out, H_out, W_out)
+        else:
+            raise ValueError(
+                "{}-dimensional Maxpool. is not implemented.".format(self.N)
+            )
```

Comment:
I think we can drop that statement, since the proper range of `self.N` is already checked in the constructor